var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -184,6 +184,14 @@ class Cp2k(MakefilePackage, CMakePackage, CudaPackage, ROCmPackage):
             description="Use CUBLAS for general matrix operations in DBCSR",
         )
 
+    # fix a memory leak introduced when unified memory support was added.
+    with when("+rocm") or when("+hip_backend_cuda"):
```

Comment:
I know it reads well, but did you check whether this context manager works with `or`? :slightly_smiling_face: